### PR TITLE
ArgoCD guide

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

First pass at the [ArgoCD](https://argoproj.github.io/argo-cd/) CI/CD guide.

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation], [examples], or [release notes].
- [x] Updated the RBAC permissions in [`clusterrole.yaml`] to include any new types.

[documentation]: https://github.com/crossplaneio/crossplane/tree/master/docs
[examples]: https://github.com/crossplaneio/crossplane/tree/master/cluster/examples
[release notes]: https://github.com/crossplaneio/crossplane/tree/master/PendingReleaseNotes.md
[`clusterrole.yaml`]: https://github.com/crossplaneio/crossplane/blob/master/cluster/charts/crossplane/templates/clusterrole.yaml

### CODE REVIEW

File: docs/workflow.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+title: "Workflow"
```
Comment (prasek) [inline docs/workflow.md]:
```suggestion
CD Workflow
```

File: docs/workflow/argocd.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+For this guide we will assume that an organization wants to enable provisioning
+of managed service on [GCP], [AWS], and [Azure]. To install Crossplane along
+with the GCP, AWS, and Azure stacks, please see the [installation guide].
+
+## Set Up an Infrastructure Pipeline
+
+Now that ArgoCD and all Crossplane components are installed, it is time to build
+out our infrastructure environments. 
+
+### Create an Infrastructure Repository
```
Comment (prasek) [inline docs/workflow/argocd.md]: would add a getting started section to workflow.md basics, then link to that section from here (to orient new users):
### Environment Configuration
 * brief intro to repo layout
 * doing basic environment customization [patch.yaml](https://github.com/hasheddan/crosscd/blob/master/classpacks/services-guides/gcp/patch.yaml)
 * `kubectl apply -k` apply it

Comment (hasheddan) [reply docs/workflow/argocd.md]: @prasek thinking about making this a little more basic (i.e. more like the demo on TBS today). What do you think? How much of this do you still want moved to `workflow.md`?

Comment (prasek) [reply docs/workflow/argocd.md]: @hasheddan yes, the TBD demo was awesome! 

suggest going with something similar to https://github.com/crossplaneio/tbs/tree/master/episodes/5/assets/infra plus:
 - infra classes for each kind
 - env (dev, test) patch for kustomizing `region` and `network`

Then do this for each cloud:
```text
aws
└── app   # wordpress examples
└── env   # environments
       └── dev # ../../infra base + dev patch.yaml
       └── test # ../../infra base + test patch.yaml
└── infra  # classes for all kinds plus secure networking / conn

gcp
└── app
└── env 
       └── dev
       └── test
└── infra

azure
└── app
└── env 
       └── dev
       └── test
└── infra
```

**Comment by hasheddan** [comment]
@prasek I have updated per our conversations and I believe this should be pretty close to RTM. A couple of notes:
- AWS setup is linked but I am still nailing down the configuration in https://github.com/hasheddan/crosscd, no changes will need to be made to this guide after it is complete though
- Azure is excluded due to the vnet rule situation that we are still designing a solution for

**Comment by jbw976** [comment]
@hasheddan would you like to make the https://github.com/hasheddan/crosscd repo part of the crossplaneio org?

**Comment by hasheddan** [comment]
@jbw976 that would be great!

**Comment by jbw976** [comment]
i wonder how best to do that.  do you want to add me as a contributor (admin?) to your project so I can move it to the crossplaneio org?

**Comment by hasheddan** [comment]
@jbw976 that sounds good, let me finish up this AWS bit first :)

File: docs/workflow.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+title: "CD Workflow"
+toc: true
+weight: 590
+indent: false
+---
+
+# CI/CD Using Crossplane
```
Comment (prasek) [inline docs/workflow.md]:
```suggestion 
Continuous Delivery Using Crossplane
```

File: docs/workflow.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Crossplane enables continuous integration and continuous delivery by integrating
```
Comment (prasek) [inline docs/workflow.md]:
```suggestion 

Crossplane enables continuous delivery by integrating
```

File: docs/workflow.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+with platforms such as [ArgoCD], [Jenkins], [Gitlab], and many more. The
+separation of concern model allows for individuals and organizations to define
+their infrastructure universe for different environments, then deploy
+applications that make use of that infrastructure, all via a [GitOps] workflow.
+
+These guides serve to demonstrate common scenarios where it is desirable to
+manage all infrastructure and applications as code.
+
```
Comment (prasek) [inline docs/workflow.md]: suggest adding a link to ArgoCD Guide here

once you land this I’ll add a GitLab ADO guide page and link here as well.

File: docs/workflow/argocd.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+Now we need to set up `dev` and `prod` environment resources for GCP in our
+`infra` project. To do so, go to `Applications > New Application` and configure
+as shown below or as appropriate with your repository:
+
+![alt argo-infra-app-1](argo-infra-dev-app-1.png)
+
+![alt argo-infra-app-2](argo-infra-dev-app-2.png)
+
+After creating the application, you should see healthy status and a
```
Comment (prasek) [inline docs/workflow/argocd.md]:
```suggestion 
After creating the dev and prod environments in our infra project as ArgoCD applications, you should see healthy status and a
```

File: docs/workflow/argocd.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+visualization of all resources that were created:
+
+![alt argo-infra-app-viz](argo-infra-dev-app-viz.png)
+
+You will notice that resource classes have both a `dev` and `prod` version
+created, while the network resources (i.e. `Network`, `Subnetwork`,
+`GlobalAddress`, `Connection`) have only one version and it is a managed
+resource. These resources do not support dynamic provisioning and do not have
```
Comment (prasek) [inline docs/workflow/argocd.md]: would expect separate dev and prod networks/subnets ... this is probably fine for the example, but would be a stronger example with separate IMO

File: docs/workflow/argocd.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+visualization. You can click on the `Service` resource to get the external IP
+address of the Wordpress application when it becomes available.
+
+![alt argo-wordpress-service](argo-wordpress-service.png)
+
+If you navigate to the IP address, you should see the Wordpress start page.
+
+![alt argo-wordpress-landing](argo-wordpress-landing.png)
+
+## Customizing
```
Comment (prasek) [inline docs/workflow/argocd.md]:
```suggestion
## Portable App Deployment to Multiple Clouds
```

File: docs/workflow/argocd.md (hunk 7 above)
Comment (hasheddan) [reply docs/workflow/argocd.md]: I definitely prefer the strongest example :) However, the one drawback I see to this is that we are creating the network resources statically so we will provision a prod network in addition whether a use wants to use this guide to deploy prod and dev or not. What would you think about just adding a sentence here to the effect of "In a production scenario, you may want to set up separate networking resources for prod and dev."?

Comment (hasheddan) [reply docs/workflow/argocd.md]: I have added a similar sentence in the most recent revision

File: docs/workflow/argocd.md [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+
+## Clean Up
+
+To clean up all of our application components, you can simply delete each
+application and those resources will be removed from your cluster.
+
+## Next Steps
+
+Check out one of the other continuous delivery guides or customize your
+environments for your specific use case!
```
Comment (hasheddan) [inline docs/workflow/argocd.md]: @prasek do you want to add a link to Gitlab example here when it is ready?

Comment (prasek) [reply docs/workflow/argocd.md]: @hasheddan yes, will do after you land this with a follow-on PR, thanks!

File: docs/workflow/argocd.md (hunk 7 above)
Comment (prasek) [reply docs/workflow/argocd.md]: sounds good.

File: docs/workflow/argocd.md (hunk 9 above)
Comment (prasek) [reply docs/workflow/argocd.md]: actually will add it to the workflow.md

**Pull request merged**